### ISSUE TITLE
Travel Auth: Multi destination no ability to add row

### ISSUE THREAD

**Issue by dpdavids** [ytgov/travel-authorization#119]
**Describe the bug**
On the travel auth details page (https://travel-auth-dev.ynet.gov.yk.ca/my-travel-requests/217/details/edit) when a user selects the trip type: Multi-Destination.  The button to add a row (from, to , date, time, travel, accoms) is not available.  There is no way to add any of the trip details.

**To Reproduce**
1. Go to '+Travel Authorization' on the my-travel-requests page
2. Enter the purpose information in the Purpose region
3. in the Details region switch the Trip Type: Multi-Destination
4. See issue (missing button)

**Expected behavior**
On Trip Type: Multi-Destination select I would expect to see the first row added with (from/to/date/time/etc.) and a button to add more rows to further define my travel.

**Screenshots**
<img width="691" alt="image" src="https://github.com/user-attachments/assets/a267a8ce-33eb-4ddf-a5d7-03f59f31a27d">

**Desktop (please complete the following information):**

 - Browser Chrome
 - Version v2024.7.29.1

**Comment by klondikemarlen** [comment]
# Reflections

1. Current UI and data modeling don't support an arbitrary number of steps so this will require some heavy rebuilding.
2. Current travel stops are modeled with two models "stops" (legacy) and "trip_segments". I believe it is time to migrate fully to trip segments before trying to solve this issue as they were created to better handle multi-destination validation.

**Comment by dpdavids** [comment]
The end date does not populate at the top of the screen when an additional leg is added.  The days are not calculating properly when an additional leg is added.

https://travel-auth-dev.ynet.gov.yk.ca/travel-requests/253/details/manage

![Image](https://github.com/user-attachments/assets/0114dcf6-509f-4baf-82f0-081e8e51324b)

**Comment by klondikemarlen** [comment]
Good catch! That's my mistake, I wasn't sure if the initial day should be counted.

**Comment by klondikemarlen** [comment]
The current header isn't reactive currently, but I can see about re-enabling that.
After you progress pass the step, it should update the header.
